Unsupported data dtype for FindContours in vista2d

```
2024-09-25 14:14:03,491 - INFO - --- input summary of monai.bundle.scripts.run ---
2024-09-25 14:14:03,491 - INFO - > config_file: 'configs/inference.json'
2024-09-25 14:14:03,491 - INFO - > input_dict: {'image': '/cellpose_dataset/test/014_img.png'}
2024-09-25 14:14:03,491 - INFO - > output_ext: '.png'
2024-09-25 14:14:03,491 - INFO - ---


WARNING:py.warnings:The 'warn' method is deprecated, use 'warning' instead

2024-09-25 14:14:03,492 - WARNING - Default logging file in configs/logging.conf does not exist, skipping logging.
WARNING:py.warnings:Detected deprecated name 'optional_packages_version' in configuration file, replacing with 'required_packages_version'.

WARNING:py.warnings:You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.

WARNING:py.warnings:You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.

INFO:root:Restored all variables from ./models/model.pt
INFO:ignite.engine.engine.SupervisedEvaluator:Engine run resuming from iteration 0, epoch 0 until 1 epochs
WARNING:py.warnings:Divide by zero (a_min == a_max)

WARNING:py.warnings:`torch.cuda.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cuda', args...)` instead.

2024-09-25 14:14:09,605 - INFO - File: /cellpose_dataset/test/014_img.png; Base: 014_img
2024-09-25 14:14:09,606 - INFO - Mask Shape: (512, 512); Instances: 277
2024-09-25 14:14:09,614 - INFO - Saving ./eval/014_img_trans.png
ERROR:ignite.engine.engine.SupervisedEvaluator:Current run is terminating due to exception: applying transform <monai.transforms.compose.Compose object at 0x7f99d507e830>
2024-09-25 14:14:09,615 - ERROR - Exception: applying transform <monai.transforms.compose.Compose object at 0x7f99d507e830>
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/monai/transforms/transform.py", line 141, in apply_transform
    return _apply_transform(transform, data, unpack_items, lazy, overrides, log_stats)
  File "/usr/local/lib/python3.10/dist-packages/monai/transforms/transform.py", line 98, in _apply_transform
    return transform(data, lazy=lazy) if isinstance(transform, LazyTrait) else transform(data)
  File "/workspace/bundles/vista2d/scripts/components.py", line 254, in __call__
    contours, _ = cv2.findContours(label, cv2.RETR_LIST, cv2.CHAIN_APPROX_SIMPLE)
cv2.error: OpenCV(4.7.0) /opencv-4.7.0/modules/imgproc/src/contours.cpp:195: error: (-210:Unsupported format or combination of formats) [Start]FindContours supports only CV_8UC1 images when mode != CV_RETR_FLOODFILL otherwise supports CV_32SC1 images only in function 'cvStartFindContours_Impl'


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/monai/transforms/transform.py", line 141, in apply_transform
    return _apply_transform(transform, data, unpack_items, lazy, overrides, log_stats)
  File "/usr/local/lib/python3.10/dist-packages/monai/transforms/transform.py", line 98, in _apply_transform
    return transform(data, lazy=lazy) if isinstance(transform, LazyTrait) else transform(data)
  File "/usr/local/lib/python3.10/dist-packages/monai/transforms/compose.py", line 335, in __call__
    result = execute_compose(
  File "/usr/local/lib/python3.10/dist-packages/monai/transforms/compose.py", line 111, in execute_compose
    data = apply_transform(
  File "/usr/local/lib/python3.10/dist-packages/monai/transforms/transform.py", line 171, in apply_transform
    raise RuntimeError(f"applying transform {transform}") from e
RuntimeError: applying transform <scripts.components.SaveTiffExd object at 0x7f99d5f9ba00>

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/ignite/engine/engine.py", line 1068, in _run_once_on_dataset_as_gen
    self.state.output = self._process_function(self, self.state.batch)
  File "/usr/local/lib/python3.10/dist-packages/monai/engines/evaluator.py", line 349, in _iteration
    engine.fire_event(IterationEvents.MODEL_COMPLETED)
  File "/usr/local/lib/python3.10/dist-packages/ignite/engine/engine.py", line 449, in fire_event
    return self._fire_event(event_name)
  File "/usr/local/lib/python3.10/dist-packages/ignite/engine/engine.py", line 425, in _fire_event
    func(*first, *(event_args + others), **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/monai/engines/workflow.py", line 226, in _run_postprocessing
    engine.state.batch[i], engine.state.output[i] = engine_apply_transform(b, o, posttrans)
  File "/usr/local/lib/python3.10/dist-packages/monai/engines/utils.py", line 336, in engine_apply_transform
    transformed_data = apply_transform(transform, data)
  File "/usr/local/lib/python3.10/dist-packages/monai/transforms/transform.py", line 171, in apply_transform
    raise RuntimeError(f"applying transform {transform}") from e
RuntimeError: applying transform <monai.transforms.compose.Compose object at 0x7f99d507e830>
ERROR:ignite.engine.engine.SupervisedEvaluator:Engine run is terminating due to exception: applying transform <monai.transforms.compose.Compose object at 0x7f99d507e830>
2024-09-25 14:14:09,616 - ERROR - Exception: applying transform <monai.transforms.compose.Compose object at 0x7f99d507e830>
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/monai/transforms/transform.py", line 141, in apply_transform
    return _apply_transform(transform, data, unpack_items, lazy, overrides, log_stats)
  File "/usr/local/lib/python3.10/dist-packages/monai/transforms/transform.py", line 98, in _apply_transform
    return transform(data, lazy=lazy) if isinstance(transform, LazyTrait) else transform(data)
  File "/workspace/bundles/vista2d/scripts/components.py", line 254, in __call__
    contours, _ = cv2.findContours(label, cv2.RETR_LIST, cv2.CHAIN_APPROX_SIMPLE)
cv2.error: OpenCV(4.7.0) /opencv-4.7.0/modules/imgproc/src/contours.cpp:195: error: (-210:Unsupported format or combination of formats) [Start]FindContours supports only CV_8UC1 images when mode != CV_RETR_FLOODFILL otherwise supports CV_32SC1 images only in function 'cvStartFindContours_Impl'
```

--- a/models/vista2d/configs/metadata.json
+++ b/models/vista2d/configs/metadata.json
@@ -1,7 +1,8 @@
 {
     "schema": "https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/meta_schema_20240725.json",
-    "version": "0.2.8",
+    "version": "0.2.9",
     "changelog": {
+        "0.2.9": "fix unsupported data dtype in findContours",
         "0.2.8": "remove relative path in readme",
         "0.2.7": "enhance readme",
         "0.2.6": "update tensorrt benchmark results",

--- a/models/vista2d/scripts/components.py
+++ b/models/vista2d/scripts/components.py
@@ -251,6 +251,7 @@ class SaveTiffExd(MapTransform):
                 cv2.imwrite(output_filepath, image)
                 logger.info(f"Overlay Masks: Saving {output_filepath}")
             else:
+                label = cv2.convertScaleAbs(label, alpha=255.0/label.max())
                 contours, _ = cv2.findContours(label, cv2.RETR_LIST, cv2.CHAIN_APPROX_SIMPLE)
                 polygons.extend(self.to_polygons(contours))
 

--- a/models/vista2d/scripts/components.py
+++ b/models/vista2d/scripts/components.py
@@ -251,7 +251,7 @@ class SaveTiffExd(MapTransform):
                 cv2.imwrite(output_filepath, image)
                 logger.info(f"Overlay Masks: Saving {output_filepath}")
             else:
-                label = cv2.convertScaleAbs(label, alpha=255.0/label.max())
+                label = cv2.convertScaleAbs(label, alpha=255.0 / label.max())
                 contours, _ = cv2.findContours(label, cv2.RETR_LIST, cv2.CHAIN_APPROX_SIMPLE)
                 polygons.extend(self.to_polygons(contours))
 

